integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java

```diff
@@ -71,14 +71,14 @@
   private AMRMClientAsync<ContainerRequest> mRMClient;
   /** Client to talk to Node Manager */
   private NMClient mNMClient;
-  /** Whether a container for Tachyon master is allocated */
-  private volatile boolean mMasterContainerAllocated;
-  /** Num of allocated worker containers */
-  private volatile AtomicInteger mNumAllocatedWorkerContainers;
+  /** The count starts at 1, then becomes 0 when we allocate a container for the Tachyon master */
+  private CountDownLatch mMasterContainerAllocated;
```

Comment:
I think it's the best tool for the job, but I'm open to other suggestions. The docs for `CountDownLatch` say 

> A CountDownLatch is a versatile synchronization tool and can be used for a number of purposes. A CountDownLatch initialized with a count of one serves as a simple on/off latch, or gate: all threads invoking await wait at the gate until it is opened by a thread invoking countDown()